Lib/test/support/__init__.py

```diff
@@ -649,20 +654,29 @@ def wrapper(*args, **kw):
     return decorator
 
 
-def requires_hashdigest(digestname):
+def requires_hashdigest(digestname, openssl=None, usedforsecurity=True):
     """Decorator raising SkipTest if a hashing algorithm is not available
 
     The hashing algorithm could be missing or blocked by a strict crypto
     policy.
 
+    If 'openssl' is True, then the decorator checks that OpenSSL provides
+    the algorithm. Otherwise the check falls back to built-in
+    implementations. The usedforsecurity flag is passed to the constructor.
+
     ValueError: [digital envelope routines: EVP_DigestInit_ex] disabled for FIPS
     ValueError: unsupported hash type md4
     """
     def decorator(func):
```

Comment:
"import test.support" became an import nightmare: it imports way too many modules :-(

Would it be possible to import _hashlib here?

```
try:
    import _hashlib
except ImportError:
    _hashlib = None
```

Reply:
It's possible but it doesn't make much sense IMO. The function would have to try/import/except for every decorator call. The support helper package is already a kitchen sink.

Reply:
> It's possible but it doesn't make much sense IMO.

"import test.support" is used by regrtest, our test runner, and almost all tests. Reducing imports should make tests faster and more "isolated".

> The function would have to try/import/except for every decorator call. The support helper package is already a kitchen sink.

Packages are put in sys.modules and so only the first import is slow.

We should split test.support monster into subpackages like test.support.sslutils.